app/controllers/recipients_controller.rb

```diff
@@ -0,0 +1,41 @@
+class RecipientsController < ApplicationController
+  
+  before_action :set_message
+
+  # GET /recipients/1
+  def show
+    if @message.authenticated?
+      @message.delete()
+    else
+      redirect_to action: 'authenticate'
+    end
+  end
+
+  # GET /recipients/1/authenticate
+  def authenticate
+  end
+
+  # POST /recipients/1
+  def authenticatedShow
+    paramHash = message_params
```

Comment:
no need to store this off in a new variable.  we can just reference `strong_params[:password]`.